lib/os.js

```diff
@@ -23,16 +23,20 @@ exports.platform = function() {
 };
 
 exports.tmpdir = function() {
+  var path;
   if (isWindows) {
-    return process.env.TEMP ||
+    path = process.env.TEMP ||
            process.env.TMP ||
            (process.env.SystemRoot || process.env.windir) + '\\temp';
   } else {
-    return process.env.TMPDIR ||
+    path = process.env.TMPDIR ||
            process.env.TMP ||
            process.env.TEMP ||
            '/tmp';
   }
+  if (/[\\\/]$/.test(path))
+    path = path.substr(0, path.length - 1);
```

Comment:
Maybe this:

```js
path = require('path').normalize(path);
if (/[\\\/]$/.test(path))
  path = path.substr(0, path.length - 1);
```

(And of course putting the require call outside the function)